### PR TITLE
Pré-processamento de dataset gerado

### LINKED ISSUE #12: Normalizar e Escalar os valores de um novo dataset
## Descrição

Ao gerar novos datasets, tomamos como base a criação de valores randomizados que seguem diversos padrões que já foram aderidos pelo framework, ~até o momento padrões de geração do sci-kit learn~, todavia esses valores por muitas vezes não condizem exatamente com os valores de datasets reais. 

Tome de fato o dataset `iris`, para este dataset temos 4 informações válidas para a classicação, que são o `Sepal.Length`, `Sepal.Width`, `Petal.Length` e `Petal.Width`. Para tal, faremos um sumário de cada atributo, vejamos:

```
##   Sepal.Length     Sepal.Width    Petal.Length     Petal.Width   
##  Min.   :4.300   Min.   :2.000   Min.   :1.000   Min.   :0.100  
##  1st Qu.:5.100   1st Qu.:2.800   1st Qu.:1.600   1st Qu.:0.300  
##  Median :5.800   Median :3.000   Median :4.350   Median :1.300  
##  Mean   :5.843   Mean   :3.057   Mean   :3.758   Mean   :1.199  
##  3rd Qu.:6.400   3rd Qu.:3.300   3rd Qu.:5.100   3rd Qu.:1.800  
##  Max.   :7.900   Max.   :4.400   Max.   :6.900   Max.   :2.500  
```

Observe que os valores obedecem o seguinte padrão de normalização do dataset e para tal, se queremos gerar um dataset sintético semelhante ao real, devemos observar a normalização dos valores de cada instância.

## Objetivos

- [ ] Gerar instâncias que obedeçam o valores de mínimo e máximo do dataset original;
- [ ] Continuar Garantindo o _RandomState_ para cada atributo;
- [ ] Observar a eficiência da normalização;
- [ ] Garantir a normalização e concordância do dataset sintético.

## Referências

- <https://stackoverflow.com/questions/26414913/normalize-columns-of-pandas-data-frame>

### PR DESCRIPTION
**Pull request by SteffanoP**
## Descrição Rápida

Esta Pull Request visa criar um novo módulo para o pré-processamento do dataset gerado, visando o mínimo possível para que o dataset gerado seja semelhante ao dataset original.

Essa PR também visa solucionar os problemas em #12, do qual havia a necessidade de manter o dataset escalado, seguindo as mesmas escalas de máximo e mínimo do dataset original.

Fixes #12 

## Objetivos

- [x] Gerar instâncias que obedeçam o valores de mínimo e máximo do dataset original;
- [x] Continuar Garantindo o _RandomState_ para cada atributo;
  - Mantido pelo método de escala ([`MinMaxScaler`](https://scikit-learn.org/stable/modules/generated/sklearn.preprocessing.MinMaxScaler.html)). 
- [ ] Observar a eficiência da normalização;
- [ ] Garantir a normalização e concordância do dataset sintético.

### CODE REVIEW

File: src/cbdgen-framework.py [hunk 1]
```diff
@@ -90,6 +92,12 @@
     base_df = pd.DataFrame(data=np.c_[base_dataset['data'], base_dataset['target']], columns=base_dataset['feature_names'] + ['target'])
     target = "target"
 
+    # Copying Columns names
+    df.columns = preprocess.copyColumnsNamesFrom(base_df, label_name=target)
+
+    # Scaling to original Dataset
+    df = preprocess.scaleColumnsFrom(base_df, df)
+
```
Comment (SteffanoP) [inline src/cbdgen-framework.py:100]: Durante alguns testes realizados, foi observado que o algoritmo genético não se comporta muito bem com escalas diferente das originais, muito provavelmente devido as configurações de peso e pontos de referência, logo não é interessante realizar o _Scale_ antes do processo geracional.

A sugestão aqui é o seguinte, para evitar interferências de geração e rotulação do dataset pelo algoritmo genético, é importante realizar o _Scale_ **após** a geração do dataset sintético, ou estabelecer a escala como uma opção ao usuário; pois, quando escalamos os atributos do dataset, em tese, nós não alteramos sua complexidade (isso foi observado após a geração de um dataset no formato anterior a esta PR e escalando seus valores após a rotulação, e o resultado das complexidades obtidas foram as mesmas.

**Comment by SteffanoP** [comment]
[Normalizar e Escalar os valores de um novo dataset](https://trello.com/c/44teXMtY/44-normalizar-e-escalar-os-valores-de-um-novo-dataset)

**Review by SteffanoP — COMMENTED** [review]
Mesmo após novas implementações, não foi possível constatar a eficiência do escalamento dos valores de complexidade, ainda será necessário verificar a real eficiência dessa implementação, mesmo que temporária.
